compiler/src/dmd/opover.d

```diff
@@ -517,6 +517,61 @@ Expression op_overload(Expression e, Scope* sc, EXP* pop = null)
         return result;
     }
 
+    // When no operator overload functions are found for `e`, recursively try with `alias this`
+    // Returns: `null` when still no overload found, otherwise resolved lowering
+    Expression binAliasThis(BinExp e, AggregateDeclaration ad1, AggregateDeclaration ad2)
```

Comment:
N.b. this is outside `visitBin` because I plan to untangle the intertwined cases of opBinary and opAssign, which are currently in the same `visitBin` function, so `binAliasThis` needs to be callable from multiple functions.